src/api/getWilayaByPhoneCode.js

```diff
@@ -0,0 +1,23 @@
+const projectWilaya = require('../utils/projections/wilayaProjection');
+
+const getWilayaByPhoneCode = data =>
+/**
+* Takes a phone code  or the full phone number and returns its wilaya.
+*
+* @example Get 34 wilaya
+*
+* //returns {mattricule: '6', name: 'Béjaïa', ...} because the phone code 34 belongs to Béjaïa
+* getWilayaByPhoneCode(34)
+*
+* @param { Number | String } phoneCode is a number if the latter corresponds to the wilaya's phone code and a string if's the full phone number
+* @param {String[]} projection a list of wilaya object attributes to keep
+* @returns { Object | undefined } Returns the target object, or undefined
+*/
+  (phoneCode, projection) => {
+    const phoneCodeVar =  typeof phoneCode === "number"
```

Comment:
calling the function with `"34"` for example will lead us to the else -> truncate the number, please fix it and add a test case to cover this, probably we may break it in the future 🙄 .  Also, what if we call this functions with `"034"`?